### DISCUSSION TITLE
negative soft limits

### DISCUSSION THREAD

**Discussion by wbendick** [fra589/grbl-Mega-5X#281] [Q&A] [answered]
I have my limit switches at the maximum end of travel for all three of my linear axis. If I home my machine and then enable soft limits I can't move any axis.

>>> $J=G21G91X-10F1000
[Error] An error was detected while sending '$J=G21G91X-10F1000': (error:15) Jog target exceeds machine travel. Jog command has been ignored. Streaming has been paused.

I assumed this was because positive values for the maximum travel variables, $130-$132 were creating a box outside of my machine's space. Setting them negative seems disallowed.

>>> $130=-3600.000
[Error] An error was detected while sending '$130=-3600.000': (error:4) Negative value received for an expected positive value. Streaming has been paused.

What is the correct way to set up soft limits for a machine that only moves in negative machine coordinates?

**Comment by wbendick** [comment, marked as answer]
I figured it out by adding some debug statements. It turned out the issue was that (unused) axis 6 had the value -0.000 stored for it's maximum travel variable, $135 which would trigger the soft limit check every time no matter which axis was moving.

**Comment by fra589** [comment]
Hi @wbendick,  
  
If you only use less than 6 axes, it will be better to define the real number of axes in config.h (`#define N_AXIS N`).  
So, the behavior you described don't will happen.  
What's more, grbl-Mega-5X will be more efficient.  
(warning, don't forget issuing the `$RST=*` grbl's command when changing the axes numbers in config.h).  
  
@++;
Gauthier.

**Reply by wbendick** [reply]
Originally I tried this, but when generating code for my 4th axis with fusion360, in addition to the a axis angles, it would specify b0 and c0. This caused grbl to fail because the axis were not defined so i just added them and left them unused.
